### PR TITLE
Add feature importance plot to LightGBM auto logging

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

Follow-up for #2275. Add feature importance plot to LightGBM auto logging.

## How is this patch tested?

Unit tests

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [x] API
- [ ] REST-API
- [ ] Examples
- [x] Docs
- [x] Tracking
- [ ] Projects
- [ ] Artifacts
- [x] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [x] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/lightgbm.py
Comment (harupy) [inline mlflow/lightgbm.py]: Maybe this functions should be moved to [`mlflow/utils`](https://github.com/mlflow/mlflow/tree/master/mlflow/utils) so that we can reuse it on `xgboost.autolog`.

**Comment by harupy** [comment]
LightGBM has a function named [`plot_importance`](https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.plot_importance.html#lightgbm-plot-importance), but this function:
- doesn't provide an option to normalize feature importance.
- doesn't adjust the figure height when the number of features to plot is large.

File: mlflow/lightgbm.py
Comment (smurching) [inline mlflow/lightgbm.py]: One question around these parameters (I'm curious since I'm not a lightgbm user myself) - how commonly do you imagine users would want to configure these? If it's rare, or if the defaults would work reasonably well, we could just remove these parameters for now & wait for user feedback (& you might even be able to provide this feedback yourself if you've been using lightgbm) that suggests we should include them, as it's good to default to keeping APIs simple where possible.

Comment (harupy) [reply mlflow/lightgbm.py]: Agreed, will remove them for now.

File: mlflow/lightgbm.py
Comment (smurching) [inline mlflow/lightgbm.py]:
```suggestion
        def log_feature_importance_plot(features, importance, importance_type):
```

Just to make the name a bit clearer

File: mlflow/lightgbm.py
Comment (smurching) [inline mlflow/lightgbm.py]: Looking at Travis tests, it seems generating the feature importance plot via matplotlib can fail if the user's environment isn't properly configured (see [this SO post](https://stackoverflow.com/questions/37604289/tkinter-tclerror-no-display-name-and-no-display-environment-variable)). Given that, we should probably try-catch this plotting logic. We can initialize a logger object at the top of our module (see analogous [mlflow.tensorflow code](https://github.com/mlflow/mlflow/blob/8447c4201ae7306d69b692f5e634c7eb59cf1462/mlflow/tensorflow.py#L47) for example) & then use it to log any plotting failures that occur:

```suggestion
      try:
            log_feature_importance_plot(features, importance, imp_type)
      except Exception as e:
            _logger.exception("Failed to log feature importance plot. LightGBM autologging will ignore the failure and continue. Exception: ")
```

Comment (smurching) [reply mlflow/lightgbm.py]: We likely also need to configure matplotlib to not use the Tkinter backend in our Travis tests, as described in the above SO post.

File: mlflow/lightgbm.py
Comment (smurching) [inline mlflow/lightgbm.py]:
```suggestion
                _logger.exception('Failed to log feature importance plot. LightGBM autologging '
```

Micronit: just making sure there are spaces between words in the error message

**Review by smurching — APPROVED** [review]
Tiny nit, LGTM after that's addressed

File: mlflow/lightgbm.py
Comment (harupy) [reply mlflow/lightgbm.py]: Oh, thanks for the catch!

Comment (harupy) [reply mlflow/lightgbm.py]: Fixed it.

**Review by smurching — APPROVED** [review]
LGTM

**Pull request merged**
